salt/_pillar/metalk8s_nodes.py

```diff
@@ -92,9 +96,40 @@ def ext_pillar(minion_id, pillar, kubeconfig):
             for node in node_list.items
         )
 
+        custom_object = []
+
+        try:
+            object_response = customObjectsApi.list_cluster_custom_object(
+                group="storage.metalk8s.scality.com", version="v1alpha1",
+                plural="volumes"
+            )
+            if not object_response:
+                custom_object = __utils__['pillar_utils.errors_to_dict']([
+                    'Unable to retrieve node volume list'
+                ])
+            for node in node_list.items:
+                for k, v in object_response.items():
+                    if k == "items":
+                        for items in v:
+                            volume_type = list(items['spec'].keys())[0]
+                            result_vol_object = {
+                             'volume_name': items['metadata']['name'],
+                             'volume_uid': items['metadata']['uid'],
+                             'volume_type': volume_type,
+                             'volume_capacity': items['spec'][
+                                 'sparseLoopDevice']['size'],
+                             'volume_class': items['spec']['storageClassName']}
```

Comment:
This won't be sufficient. Every volume object in the pillar will need to contain the values we care about retrieved from the storage class (hint: add a cache when looking up StorageClasses). Just the name is not useful.

Reply:
@NicolasT it would be helpful if you could provide a pre-computed list of  `storageclass` attributes we will like to add to each individual volume pillar. `mount_options`, `reclaim_policy`  does not seem sufficient to me.

Reply:
I can't :) @slaperche-scality ? This is mainly about the file-system to be used, the `mkfs` options,... stored in annotations.

Reply:
We won't use annotation but `parameters` to put our fields (`fsType` and `mkfsOptions`):

Here is an example:

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: standard
provisioner: kubernetes.io/no-provisioner
parameters:
  fsType: ext4
  mkfsOptions: '["-m", "0", "-b", "1024", "-E", "stride=16,stripe-width=64"]'
reclaimPolicy: Retain
mountOptions:
  - debug
volumeBindingMode: Immediate
```